rad/src/reducers/average.rs

```diff
@@ -1,24 +1,167 @@
-use crate::error::RadError;
-use crate::types::{array::RadonArray, float::RadonFloat, RadonType, RadonTypes};
-
-use std::ops::Div;
+use crate::{
+    error::RadError,
+    operators::array::transpose,
+    types::{array::RadonArray, float::RadonFloat, RadonType, RadonTypes},
+};
+use std::{mem::discriminant, ops::Div};
 
 pub fn mean(input: &RadonArray) -> Result<RadonTypes, RadError> {
     let value = input.value();
+    let value_len = value.len();
+    let inner_type = input.inner_type();
 
-    // Sum all numeric values
-    let (sum, count) = value
-        .iter()
-        .fold((0f64, 0f64), |(sum, count), item| match item {
-            RadonTypes::Float(f64_value) => (sum + f64_value.value(), count + 1f64),
-            // TODO: implement RadonInteger branch here
-            // RadonTypes::Integer(i64_value) => (sum + acc + i64_value.value() as f64, count + 1f64),
-            // Skip any non-numeric RadonType
-            _ => (sum, count),
+    // Float case
+    if inner_type == discriminant(&RadonTypes::Float(RadonFloat::default())) {
+        let sum = value.iter().try_fold(0f64, |sum, item| match item {
+            RadonTypes::Float(f64_value) => Ok(sum + f64_value.value()),
+            _ => Err(RadError::MismatchingTypes),
         });
+        let sum = sum?;
+
+        // Divide sum by the count of numeric values that were summed
+        let mean_value: f64 = sum.div(value_len as f64);
```

Comment:
What should happen if `value_len == 0`?

Currently, it returns `NaN`. Add test:

```rust
#[test]
fn test_operate_reduce_mean_empty() {
    let input = RadonArray::from(vec![]);
    let output = mean(&input).unwrap();
    assert_eq!(
        output,
        RadonTypes::from(RadonFloat::from(std::f64::NAN))
    );
}
```